### ISSUE TITLE
bug: vue, incorrectly routes back when tab is undefined

### ISSUE THREAD

**Issue by snowwolfjay** [ionic-team/ionic-framework#25705]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x
- [ ] Nightly

### Current Behavior

We use createMemoryHistory to build the router, android is ok.
But when test in ios , strange things happen, need forbid swipe back function which is so bad :
Suppose there is an app with three page, home -> child1 -> child2 
1. launch home page, then nav to child1 ,then nav to child2
2. swipe back twice to home page,
3. console log has a error, and then router breaks

Our app has almost 50 pages, and after swipe back twice, router works like chaos, it randomly navigate to some page... or do nothing ...

### Expected Behavior

When use memory history and swipe back twice on IOS , router not break

### Steps to Reproduce

1.  open cmd , ionic start create new project
2.  pick vue  + list template
3.  create a new page, and add a button at detail page which can navigate to the new page.
4.  change the router/index.ts ,replace createWebHistory to  createMemoryHistory 
5.  ionic serve to preview at browser with devtool open and choose the ios device 
6.  nav to detail page then to new page
7. swipe back and swipe back to home page
8. console log print the error.

----
if need test the random nav need create more page and add entry to home page, 
after step 8 , try nav to other page can see the router navigate randomly ..

### Ionic Info

Ionic:

   Ionic CLI       : 6.18.1 (C:\Users\by4474\AppData\Roaming\npm\node_modules\@ionic\cli)
   Ionic Framework : @ionic/vue 6.2.0

Capacitor:

   Capacitor CLI      : 3.5.1
   @capacitor/android : 3.5.1
   @capacitor/core    : 3.5.1
   @capacitor/ios     : 3.5.1

Utility:

   cordova-res : not installed globally
   native-run  : 1.6.0

System:

   NodeJS : v16.15.1 (C:\Program Files\nodejs\node.exe)
   npm    : 8.12.1
   OS     : Windows 10

### Additional Information

not now,

**Comment by liamdebeasi** [comment]
Thanks for the issue. Can you provide a minimal GitHub repo I can use to verify this behavior?

**Comment by snowwolfjay** [comment]
> Thanks for the issue. Can you provide a minimal GitHub repo I can use to verify this behavior?

https://github.com/snowwolfjay/reproduction-ionic-25705

this is a minimal repo.

**Comment by liamdebeasi** [comment]
Can you try the following dev build and let me know if it resolves the issue?

```
npm install @ionic/vue@6.2.2-dev.11659726184.13d7540f @ionic/vue-router@6.2.2-dev.11659726184.13d7540f
```

**Comment by snowwolfjay** [comment]
> Can you try the following dev build and let me know if it resolves the issue?
> 
> ```
> npm install @ionic/vue@6.2.2-dev.11659726184.13d7540f @ionic/vue-router@6.2.2-dev.11659726184.13d7540f
> ```

resolved, thank a lot

**Comment by liamdebeasi** [comment]
Thanks for the issue. This has been resolved via https://github.com/ionic-team/ionic-framework/pull/25732, and a fix will be available in an upcoming release of Ionic Framework.
